osu.Game/Overlays/BeatmapSet/BeatmapRulesetSelector.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets;
+using osuTK;
+using System.Linq;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapRulesetSelector : RulesetSelector
+    {
+        private BeatmapSetInfo beatmapSet;
+
+        public BeatmapSetInfo BeatmapSet
+        {
+            get => beatmapSet;
+            set
+            {
+                if (value == beatmapSet)
+                    return;
+
+                beatmapSet = value;
+
+                foreach (var tab in TabContainer.TabItems.OfType<BeatmapRulesetTabItem>())
+                    tab.SetBeatmaps(beatmapSet?.Beatmaps.FindAll(b => b.Ruleset.Equals(tab.Value)));
```

Comment:
I find this naming a bit too general, since as far as I can see the tab does not hold the actual beatmaps, just a per-ruleset count - how about something like `SetCountForRuleset`/`CountBeatmapsForRuleset` (if my other suggestion is applied)?

Reply:
as i said down there, the data required for this tab is the beatmaps of it's value (ruleset), i'm concerned whether to change to what we actually need now (`SetBeatmapsCount`) or keep it as is..

Reply:
wouldn't it be more in-line with the rest of our code to pass the beatmapset down (whether bindable or property) and have the count or whatever computed within the tab?